dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/TableAlteror.java

```diff
@@ -689,6 +689,9 @@ public void execute()
         executeSql( "DELETE FROM userroleauthorities WHERE authority='F_PROGRAM_ATTRIBUTE_UPDATE'" );
         executeSql( "DELETE FROM userroleauthorities WHERE authority='F_PATIENT_DATAVALUE_UPDATE'" );
 
+        executeSql( "insert into userroleauthorities" +

```

Comment:
1) This cannot be run every time. Imagine if someone remove the "see message module" auth for someone that has see dashboard, then the auth will be put back the next time. We need to move this to an external upgrade script.

2) Seems to be a missing space after insert and select.